packages/frontend-2/components/dashboards/share/Dialog.vue

```diff
@@ -0,0 +1,152 @@
+
+const dashboardsDialogSharePermissionsQuery = graphql(`
+  query DashboardsSharePermissions($id: String!) {
+    dashboard(id: $id) {
+      id
+      shareLink {
+        id
+        content
+        revoked
+      }
+      permissions {
+        canCreateToken {
+          ...FullPermissionCheckResult
+        }
+      }
+    }
+  }
+`)
+
+const dashboardsDialogShareTokenMutation = graphql(`
+  mutation DashboardsShareToken($dashboardId: String!) {
+    dashboardMutations {
+      createToken(dashboardId: $dashboardId) {
+        token
+      }
+    }
+  }
+`)
+
+const dashboardsDialogShareEnableTokenMutation = graphql(`
+  mutation DashboardsShareEnableToken($input: DashboardShareInput!) {
+    dashboardMutations {
+      enableShare(input: $input)
+    }
+  }
+`)
+
+const dashboardsDialogShareDisableTokenMutation = graphql(`
+  mutation DashboardsShareDisableToken($input: DashboardShareInput!) {
+    dashboardMutations {
+      disableShare(input: $input)
+    }
+  }
+`)
+
+const props = defineProps<{
+  workspaceSlug: MaybeNullOrUndefined<string>
+  dashboardId: MaybeNullOrUndefined<string>
+}>()
+
+const open = defineModel<boolean>('open', { required: true })
+
+const { result, onResult, refetch } = useQuery(
+  dashboardsDialogSharePermissionsQuery,
+  () => ({
+    id: props.dashboardId || ''
+  })
+)
+const { mutate: createToken } = useMutation(dashboardsDialogShareTokenMutation)
+const { mutate: disableToken } = useMutation(dashboardsDialogShareDisableTokenMutation)
+const { mutate: enableToken } = useMutation(dashboardsDialogShareEnableTokenMutation)
+
+const isRevoked = computed(() => result.value?.dashboard?.shareLink?.revoked)
+const shareLink = computed(() => result.value?.dashboard?.shareLink)
+const shareUrl = computed(() => {
+  if (!shareLink.value?.id || !props.workspaceSlug || !props.dashboardId) return ''
+
+  const url = new URL(
+    dashboardRoute(props.workspaceSlug, props.dashboardId),
+    window.location.toString()
+  )
+  url.searchParams.set('dashboardToken', shareLink.value.content)
+
+  return url.toString()
+})
+const canCreateToken = computed(
+  () => result.value?.dashboard?.permissions?.canCreateToken?.authorized
+)
+const enablePublicUrl = computed({
+  get: () => !isRevoked.value && !!shareLink.value?.id,
+  set: (value: boolean) => {
+    onEnablePublicUrl(value)
+  }
+})
+const dashboardUrl = computed(() => {
+  if (!props.workspaceSlug || !props.dashboardId) return ''
+
+  return new URL(
+    dashboardRoute(props.workspaceSlug, props.dashboardId),
+    window.location.toString()
+  ).toString()
+})
+
+const onEnablePublicUrl = async (value: boolean) => {
+  if (!props.dashboardId) return
+
+  if (value) {
+    // If enabling and no share link exists, create one first
+    if (!shareLink.value?.id) {
+      await createToken({ dashboardId: props.dashboardId })
+      // Create token doesnt return anything, so we need to refetch
+      await refetch()
```

Comment:
@gjedlicska the createToken mutation doesnt give me the ID, so I need to refetch it to get it, its fine for now, but later it would be nice if the mutations also returns the ID so apollo automatically updates the cache so its always up to date